src/Monzo.lua

```diff
@@ -107,184 +104,214 @@ function InitializeSession2 (protocol, bankCode, step, credentials, interactive)
-	local accounts = {}
-	for key, account in pairs(monzoAccountsResponse) do
-		accounts[#accounts+1] = {
-			-- String name: Bezeichnung des Kontos
-			name = accountNameForMonzoAccount(account),
-			-- String owner: Name des Kontoinhabers
-			owner = account.description,
-			-- String accountNumber: Kontonummer
-			accountNumber = email,
-			-- String subAccount: Unterkontomerkmal
-			subAccount = account.id,
-			-- Boolean portfolio: true für Depots und false für alle anderen Konten
-			portfolio = false,
-			-- String bankCode: Bankleitzahl
-			-- String currency: Kontowährung
-			currency = "GBP",
-			-- String iban: IBAN
-			-- String bic: BIC
-			-- Konstante type: Kontoart; 
-			type = accountTypeForMonzoAccountType(account.type)
-		}
-	end
+function ListAccounts(knownAccounts)
+  isInitialSetup = true
+  local monzoAccountsResponse = queryPrivate("accounts").accounts
+  local accounts = {}
+  for key, account in pairs(monzoAccountsResponse) do
+    accounts[#accounts + 1] = {
+      -- String name: Bezeichnung des Kontos
+      name = accountNameForMonzoAccountType(account.type),
+      -- String owner: Name des Kontoinhabers
+      owner = ownerForMonzoAccountOwners(account.owners),
+      -- String accountNumber: Kontonummer
+      accountNumber = account.account_number .. " ", -- enforces that MoneyMoney will not hide a leading zero
+      -- String subAccount: Unterkontomerkmal
+      subAccount = account.id,
+      -- Boolean portfolio: true für Depots und false für alle anderen Konten
+      portfolio = false,
+      -- String bankCode: Bankleitzahl
+      bankCode = formatSortCode(account.sort_code),
+      -- String currency: Kontowährung
+      currency = "GBP",
+      -- String iban: IBAN
+      -- String bic: BIC
+      -- Konstante type: Kontoart;
+      type = accountTypeForMonzoAccountType(account.type)
+    }
+  end
   return accounts
 end
 
-function accountNameForMonzoAccount(account)
-	if account.type == "uk_prepaid" then
-		return "Monzo Prepaid"
-	elseif account.type == "uk_retail" then
-		return "Monzo Current"
-	else 
-		return "Monzo unknown"
-	end
+function accountNameForMonzoAccountType(monzoAccountTypeString)
+  local dict = {
+    uk_prepaid = "Monzo Prepaid", -- according to monzo documentation this type does not exist: https://docs.monzo.com/#accounts
+    uk_retail = "Monzo",
+    uk_retail_joint = "Monzo Joint Account",
+  }
+
+  return dict[monzoAccountTypeString] or "Monzo Unknown"
 end
 
-function accountTypeForMonzoAccountType(monzoAccountString)
-	if monzoAccountString == "uk_prepaid" then
-		return AccountTypeGiro
-	elseif monzoAccountString == "uk_retail" then
-		return AccountTypeGiro
-	else
-		print("Unknown account type: ", monzoAccountString)
-		return AccountTypeOther
-	end
+function accountTypeForMonzoAccountType(monzoAccountTypeString)
+  local dict = {
+    uk_prepaid = AccountTypeGiro, -- according to monzo documentation this type does not exist: https://docs.monzo.com/#accounts
```

Comment:
I guess those are the initial (beta) accounts. It showed up when I initially set this up..

Reply:
do you think they could be removed?

Reply:
Sorry, I think that was badly phrased by me. I suppose those accounts still show up, if you are a beta user. Which means access to those older transactions.

I think it's nice to keep that. What do you think?

Reply:
if they still show up for you they should be definitely be kept.

i don't know the monzo history quite well. are these the accounts from when monzo was not yet a bank? a colleague lately told me, that he was forced to either update to the new account type or cancel monzo. he did the latter.

Reply:
Yeah, exactly. 
Before monzo had the banking licence, it was prepaid only. And switching to a current account disabled that prepaid account, but the transactions are still available.

Reply:
if these old accounts are still present it is good to leave it in there. thanks for clarifying.